js/layout-javascript/agenda-code.js

```diff
@@ -182,21 +182,41 @@ DynamicList.prototype.attachObservers = function() {
       if (_this.isPanning && !_this.allowClick && $(this).hasClass('open')) {
         return;
       }
-      event.stopPropagation();
+
+      if ($(event.target).hasClass('agenda-item-bookmark-holder') || $(event.target).parents('.agenda-item-bookmark-holder').length) {
+        return;
+      }
+
       var entryId = $(this).data('entry-id');
       var entryTitle = $(this).find('.agenda-item-title').text();
+
       Fliplet.Analytics.trackEvent({
         category: 'list_dynamic_' + _this.data.layout,
         action: 'entry_open',
         label: entryTitle
       });
 
-      if (_this.data.summaryLinkOption === 'link' && _this.data.summaryLinkAction) {
-        _this.openLinkAction(entryId);
-        return;
+      var beforeOpen = Promise.resolve();
+    
+      if (typeof _this.data.beforeOpen === 'function') {
+        beforeOpen = _this.data.beforeOpen({
+          config: _this.data,
+          entry: _.find(_this.listItems, { id: entryId })
```

Comment:
@hcarneiro `_.find(_this.listItems, { id: entryId })` could return `undefined` if `_this.listItems` doesn't match what's displayed to the user. This could happen if there's a custom `searchData` function like we currently do.

I therefore suggest also adding the 2 following attributes for the custom function to process:

```js
entryId: entryId,
entryTitle: entryTitle
```